fe/fe-core/src/main/java/com/starrocks/sql/analyzer/ExpressionAnalyzer.java

```diff
@@ -836,6 +839,47 @@ public Void visitInformationFunction(InformationFunction node, Scope context) {
         @Override
         public Void visitVariableExpr(VariableExpr node, Scope context) {
             try {
+                if (node.getSetType().equals(SetType.USER)) {
+                    UserVariable userVariable = session.getUserVariables(node.getName());
+                    if (userVariable == null) {
+                        node.setType(Type.STRING);
+                        node.setIsNull();
+                        return null;
+                    }
+
+                    Type variableType = userVariable.getResolvedExpression().getType();
+                    String variableValue = userVariable.getResolvedExpression().getStringValue();
+
+                    if (userVariable.getResolvedExpression() instanceof NullLiteral) {
+                        node.setType(variableType);
+                        node.setIsNull();
+                        return null;
+                    }
+
+                    node.setType(variableType);
+                    switch (variableType.getPrimitiveType()) {
```

Comment:
can be support select bitmap/hll/array/json?

Reply:
support json, bitmap/hll/array not support